pkg/avancement/service_manager.go

```diff
@@ -131,7 +143,21 @@ func (s *ServiceManager) cloneRepo(repoURL, branch string) (git.Repo, error) {
 	}
 	repo, err := s.repoFactory(repoURL, path.Join(s.cacheDir, encode(repoURL, branch)), s.debug)
 	if err != nil {
-		return nil, fmt.Errorf("failed to create repo for URL %s: %w", repoURL, err)
+		errorToDisplay := err
+		// This can contain the access token so omit it here too
+		asRealURL, parseErr := url.Parse(repoURL)
```

Comment:
what's a 'real' URL?

Reply:
Yeah I thought about this too - so, an actual Go URL object not a String, but I couldn't think of a better variable name! Will ponder over

Reply:
Agreed, yea so I figured it was just one in place and then noticed indeed it's in at least three - I'll work on this bit next thanks